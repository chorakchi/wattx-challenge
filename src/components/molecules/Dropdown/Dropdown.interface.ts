import { HTMLAttributes } from "react";

type option = {
  label: string;
  value: number;
};

export interface DropdownProps extends HTMLAttributes<HTMLElement> {
  options: Array<option>;
  value: number;
}
