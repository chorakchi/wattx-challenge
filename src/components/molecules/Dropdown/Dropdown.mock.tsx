export const mockData = {
  options: [
    { label: "10", value: 10 },
    { label: "50", value: 50 },
    { label: "all", value: 5000 },
  ],
};
