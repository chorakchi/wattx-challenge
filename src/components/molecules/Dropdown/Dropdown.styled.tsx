import styled, { css } from "styled-components";

export const StyledWrapper = styled.div`
  background-color: white;

  border-radius: 8px;
  margin: 16px 0;
  position: relative;
  width: 208px;
  height: 40px;

  .icon {
    position: absolute;
    right: 14px;
    top: 7px;
    pointer-events: none;
  }
`;

export const StyledDropdown = styled.select`
  background-color: white;
  -moz-appearance: none;
  -webkit-appearance: none;
  appearance: none;
  padding-left: 16px;
  border-radius: 8px;
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  border: 2px solid #0071ff;

  :focus {
    border-color: gray;
    box-shadow: none;
    outline: none;
  }
`;

export const StyledOption = styled.option``;
