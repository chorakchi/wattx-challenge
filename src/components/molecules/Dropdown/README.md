The value must be chosen from a predefined set of allowed values.

<iframe width= "100%" height= "400px" src="https://www.figma.com/embed?embed_host=astra&url=https://www.figma.com/file/2LokuyVBjWW20P4CcDJGra/Top-Coins?node-id=7%3A1037" ></iframe>

The "value" state with the value/onChange props combination. This state represents the value selected by the user, for instance when pressing Enter or clicking on option.

By default, the component accepts the following options structures:

### Playground:
