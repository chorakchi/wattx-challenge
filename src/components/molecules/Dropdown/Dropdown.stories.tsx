import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Dropdown } from "./Dropdown";
import { mockData } from "./Dropdown.mock";
// @ts-ignore
import readme from "./README.md";

export default {
  title: "Molecules/Dropdown",
  component: Dropdown,
  parameters: {
    backgrounds: {
      default: "light",
    },
    docs: {
      description: {
        component: readme,
      },
    },
  },
} as ComponentMeta<typeof Dropdown>;

const Template: ComponentStory<typeof Dropdown> = (args) => (
  <Dropdown {...args} />
);

export const Dropdown_ = Template.bind({});
Dropdown_.args = {
  onSelect: (id) => {
    alert(id);
  },
  ...mockData,
};
