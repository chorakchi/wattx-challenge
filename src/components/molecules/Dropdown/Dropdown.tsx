import { StyledDropdown, StyledWrapper, StyledOption } from "./Dropdown.styled";
import { DropdownProps } from "./Dropdown.interface";
import { SVG } from "components";

export const Dropdown: React.FC<DropdownProps> = ({
  options = [],
  ...props
}: DropdownProps) => {
  return (
    <StyledWrapper>
      <StyledDropdown {...props}>
        {options.map((item) => (
          <StyledOption value={item.value}>{item.label}</StyledOption>
        ))}
      </StyledDropdown>
      <SVG name="arrowDown" color="#0071FF" className="icon"></SVG>
    </StyledWrapper>
  );
};
