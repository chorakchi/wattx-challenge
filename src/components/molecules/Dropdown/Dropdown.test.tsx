import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Dropdown } from "./Dropdown";
import { mockData } from "./Dropdown.mock";

jest.mock("components/atoms/SVG/SVG", () => ({
  SVG: "svg",
}));

describe("Typography", () => {

  
  it("should render the Typography", () => {

    const subject = shallow(<Dropdown {...mockData} onSelect={()=>{}} value={1}/>);
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {

    const subject = renderer.create(<Dropdown {...mockData} onSelect={()=>{}} value={1}/>).toJSON();
    expect(subject).toMatchSnapshot();
  });
});
