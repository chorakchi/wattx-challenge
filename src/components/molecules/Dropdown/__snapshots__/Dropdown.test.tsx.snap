// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should render correctly and match to the Snapshot 1`] = `
<div
  className="Dropdownstyled__StyledWrapper-sc-1n61ek4-0 fyJHxp"
>
  <select
    className="Dropdownstyled__StyledDropdown-sc-1n61ek4-1 iOqnwJ"
    onSelect={[Function]}
    value={1}
  >
    <option
      className="Dropdownstyled__StyledOption-sc-1n61ek4-2"
      value={10}
    >
      10
    </option>
    <option
      className="Dropdownstyled__StyledOption-sc-1n61ek4-2"
      value={50}
    >
      50
    </option>
    <option
      className="Dropdownstyled__StyledOption-sc-1n61ek4-2"
      value={5000}
    >
      all
    </option>
  </select>
  <svg
    className="icon"
    color="#0071FF"
    name="arrowDown"
  />
</div>
`;
